code/modules/client/preferences_savefile.dm

```diff
@@ -168,7 +168,8 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 					underwear = "Nude"
 
 	if(pref_species && !(pref_species.id in roundstart_species))
-		pref_species = new /datum/species/human()
+		var/rando_race = pick(config.roundstart_races)
+		pref_species = new rando_race()
```

Comment:
This is the one that matters. Because before if you had humans disabled but a human character in your preferences it'd try to remedy the problem... by turning you into a human. Everything else is just updating sanity checks with the same assumption.